### PR TITLE
fix(components, protocol-designer): fix logic for disabling advanced transfer settings

### PR DESCRIPTION
**Pull request by ncdiehl11**
# Overview

Update logic for disabling mix/blowout settings depending on transfer path.
- If path is consolidate, aspirate mix field should be disabled.
- If path is distribute, dispense mix field should be disabled, and blowout field should be disabled if disposal volume field is enabled. 

Side effect of enabling disposal volume field is to disable and collapse blowout field. These changes implicate that Check and ListItem components accommodate disabled state.

Note that I don't love this fix— it breaks pattern with how we handle tooltips, but it looks like we haven't had to deal with multiple disabled tooltips on the same form field in the past. I am open to ideas on how to migrate this to a more sustainable pattern moving forward.

## Test Plan and Hands on Testing

- create or or edit transfer step to open toolbox

#### consolidate
- select 1 source and multiple destinations, and "consolidate" transfer plan
- on second page of transfer form, verify that aspirate tab mix field is disabled, and hover tooltip is correct

#### distribute
- select multiple source and 1 destinations, and "distribute" transfer plan
- on second page of transfer form, verify that dispense tab mix field is disabled, and hover tooltip is correct
- with disposal volume field unchecked, check blowout field to expand
- check disposal volume field, and verify that blowout field collapses and becomes disabled, and tooltip is correct

## Changelog

- update logic in move liquid tools for multiAspirate/Dispense paths according to scenarios in overview

## Review requests

see test plan

## Risk assessment

low

### CODE REVIEW

File: components/src/atoms/ListItem/index.tsx [hunk 1]
```diff
@@ -54,6 +63,7 @@ export function ListItem(props: ListItemProps): JSX.Element {
 
   const LIST_ITEM_STYLE = css`
     background-color: ${listItemProps.backgroundColor};
+    color: ${listItemProps.color};
```
Comment (koji) [inline components/src/atoms/ListItem/index.tsx]: most cases color is `undefined` so probably passing value would be good.

```suggestion
    color: ${listItemProps.color ?? COLORS.black90};
```

File: components/src/atoms/ListItem/index.tsx [hunk 2]
```diff
@@ -8,7 +8,12 @@ import type { StyleProps } from '../../primitives'
 
 export * from './ListItemChildren'
 
-export type ListItemType = 'error' | 'noActive' | 'success' | 'warning'
+export type ListItemType =
+  | 'error'
+  | 'noActive'
+  | 'success'
+  | 'warning'
+  | 'disabled'
```
Comment (koji) [inline components/src/atoms/ListItem/index.tsx]: nit

`disabled` is used as an attribute and we use it as a prop, so probably a different name would be better.

ex
unavailable, deactivated, etc.

**Review by koji — APPROVED** [review]
left a couple of comments but the changes look good to me.

File: components/src/atoms/ListItem/index.tsx (hunk 2 above)
Comment (jerader) [reply components/src/atoms/ListItem/index.tsx]: i agree, makes it confusing to have several "disabled"s

File: protocol-designer/src/pages/Designer/ProtocolSteps/StepForm/StepTools/MoveLiquidTools/index.tsx [hunk 3]
```diff
@@ -329,7 +352,16 @@ export function MoveLiquidTools(props: StepFormProps): JSX.Element {
...
+          tooltipText={
+            tab === 'dispense'
+              ? dispenseMixDisabledTooltipText
+              : propsForFields.aspirate_mix_checkbox.tooltipContent
+          }
+          disabled={
+            tab === 'dispense'
+              ? isDestinationTrash || formData.path === 'multiDispense'
+              : formData.path === 'multiAspirate'
+          }
```
Comment (jerader) [inline protocol-designer/src/pages/Designer/ProtocolSteps/StepForm/StepTools/MoveLiquidTools/index.tsx:364]: nit: this is ok for now but would be nice to refactor this in the future to try to get the logic upstream with `propsForFields.aspirate_mix_checkbox.tooltipContent` and `propsForFields.aspirate_mix_checkbox.disabled` to keep things organized.

File: protocol-designer/src/pages/Designer/ProtocolSteps/StepForm/StepTools/MoveLiquidTools/index.tsx [hunk 4]
```diff
@@ -416,6 +448,10 @@ export function MoveLiquidTools(props: StepFormProps): JSX.Element {
             isChecked={propsForFields.blowout_checkbox.value === true}
             checkboxUpdateValue={propsForFields.blowout_checkbox.updateValue}
             tooltipText={propsForFields.blowout_checkbox.tooltipContent}
+            disabled={
+              formData.path === 'multiDispense' &&
+              formData.disposalVolume_checkbox
+            }
```
Comment (jerader) [inline protocol-designer/src/pages/Designer/ProtocolSteps/StepForm/StepTools/MoveLiquidTools/index.tsx:454]: same here

**Review by jerader — APPROVED** [review]
lgtm

File: protocol-designer/src/pages/Designer/ProtocolSteps/StepForm/StepTools/MoveLiquidTools/index.tsx (hunk 3 above)
Comment (ncdiehl11) [reply protocol-designer/src/pages/Designer/ProtocolSteps/StepForm/StepTools/MoveLiquidTools/index.tsx:364]: I agree, especially if this logic is expected to through for several other fields. I'll ticket to address in followup

**Pull request merged**